test/karma/test-app/static-members/static-members-separate-export.tsx

```diff
@@ -0,0 +1,22 @@
+import { Component, h } from '@stencil/core';
+
+@Component({
+  tag: 'static-members-separate-export',
+})
+class StaticMembersWithSeparateExport {
```

Comment:
not to complicate things, but how does `export default` factor into this?

Reply:
(like should we have a test for that too)

Reply:
I actually had written one, and removed it (the test). Stencil won't allow default exports, even if a file consists of a single class component that is the default export.

For a project with the following component:
```tsx
import { Component, h } from '@stencil/core';

@Component({
  tag: 'my-component',
})
export default class MyComponent {
  private static myVar = "input";

  render() {
    console.log(MyComponent.myVar);
    return (<div></div>);
  }
}
```
running `npx stencil build` yields:
```console
> stencil build --docs

[23:51.2]  @stencil/core
[23:51.5]  [LOCAL DEV] v3.3.0-dev.1685644211.49532eb 🔭
[23:52.3]  build, dupe-export, prod mode, started ...
[23:52.3]  transpile started ...
[23:53.0]  transpile finished in 696 ms

[ ERROR ]  ./src/components/my-component/my-component.tsx:3:1
           To allow efficient bundling, modules using @Component() can only have a single export which is the component
           class itself. Any other exports should be moved to a separate file. For further information check out:
           https://stenciljs.com/docs/module-bundling

      L3:  @Component({                                                                                                                                                                                 
      L4:    tag: 'my-component',

[23:53.0]  build failed in 709 ms
```

I had tried a few other legal (TypeScript/JavaScript-legal) combinations, but they all yielded the same error:
```tsx
// this one has a separate export default at the end of the file
import { Component, h } from '@stencil/core';

@Component({
  tag: 'my-component',
})
export class MyComponent {
  private static myVar = "input";

  render() {
    console.log(MyComponent.myVar);
    return (<div></div>);
  }
}

export default MyComponent;
```
```tsx
// don't use export on the class declaration, but export at the very bottom
import { Component, h } from '@stencil/core';

@Component({
  tag: 'my-component',
})
class MyComponent {
  private static myVar = "input";

  render() {
    console.log(MyComponent.myVar);
    return (<div></div>);
  }
}

export default MyComponent;
```

Reply:
Ah I wasn't aware that Stencil had that limitation! Obvs fine to leave out in that case 👍